src/Aspire.Dashboard/Configuration/PostConfigureDashboardOptions.cs

```diff
@@ -35,9 +35,17 @@ public void PostConfigure(string? name, DashboardOptions options)
             options.Frontend.AuthMode = FrontendAuthMode.Unsecured;
             options.Otlp.AuthMode = OtlpAuthMode.Unsecured;
         }
+        else
+        {
+            options.Frontend.AuthMode ??= FrontendAuthMode.BrowserToken;
+            options.Otlp.AuthMode ??= OtlpAuthMode.Unsecured;
+        }
         if (options.Frontend.AuthMode == FrontendAuthMode.BrowserToken && string.IsNullOrEmpty(options.Frontend.BrowserToken))
         {
-            options.Frontend.BrowserToken = TokenGenerator.GenerateToken();
+            var token = TokenGenerator.GenerateToken();
+            // Set in configuration so the same value is used in the future.
+            _configuration[DashboardConfigNames.DashboardFrontendBrowserTokenName.ConfigKey] = token;
```

Comment:
Future? Who else reads this?

Reply:
It is run each time options change. We want the value to be consistent. I'll improve comment.